File: w10/d05/INSTRUCTORS/jungle_tree/app/controllers/monkeys_controller.rb
class MonkeysController < ApplicationController

 #  monkeys GET    /monkeys(.:format)     monkeys#index
 #        POST   /monkeys(.:format)     monkeys#create
 # monkey GET    /monkeys/:id(.:format) monkeys#show
 #        PATCH  /monkeys/:id(.:format) monkeys#update
 #        PUT    /monkeys/:id(.:format) monkeys#update
 #        DELETE /monkeys/:id(.:format) monkeys#destroy

 def index
  monkeys = Monkey.all
  render json: monkeys.to_json
 end

 def show
  monkey = Monkey.find(params[:id])
  render json: monkey.to_json
 end

 def create
  monkey = Monkey.create(monkey_params)
  render json: monkey.to_json
 end

 def update
  monkey = Monkey.find(params[:id])
  monkey.update(monkey_params)
  render json: monkey.to_json
 end

 def destroy
  monkey = Monkey.find(params[:id])
  monkey.destroy
  render json: monkey.to_json
 end

 private

 def monkey_params
  params.require(:monkey).permit(:name)
 end

end

File: w11/d01/INSTRUCTORS/node_bananas/monkeysController.js
var Monkey = require(path.join(__dirname, 'monkey'));

exports.index = function(req, res){
  Monkey.find(function(err, monkeys){
    if (err) return console.error(err);
    res.send(monkeys);
  });
}

exports.create = function(req, res){
  Monkey.create({ name: req.body.name }, function(err, monkey){
    if (err) return console.error(err);
    res.send(monkey);
  })
}

// exports.update = function(req, res){
//   Monkey.findById(req.params.id, function(err, monkey){
//   if (err) return console.error(err);
//     monkey.name = req.body.name;
//     monkey.save(function( err ){
//       if (err) return console.error(err);
//       res.send(monkey);
//     });
//   });
// }

exports.update = function(req, res){
  Monkey.findById(req.params.id, function(err, monkey){
    if (err) return console.error(err);
    monkey.update({name: req.body.name}, function(err, updated){
      if (err) return console.error(err);
      res.send(updated);
    })
  })
}


exports.delete = function(req, res){
  Monkey.findById(req.params.id, function(err, monkey){
    if (err) return console.error(err);
    monkey.remove(function( err ) {
      if (err) return console.error(err);
      return res.send({'success' : 'ok'});
    })
  });
}
